### PR TITLE
Highlight currently winning Condition in Condition Set Edit and Read-only views

### LINKED ISSUE #2749: Highlight current condition in conditionSet view
Highlight the condition that is currently winning in the conditionSet edit and read only view.

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #2749 and #2646 

**Author Checklist**

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? No
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue [hunk 1]
```diff
@@ -200,6 +206,13 @@ export default {
         },
         setEditState(isEditing) {
             this.isEditing = isEditing;
+            if (this.isEditing) {
```
Comment (nikhilmandlik) [inline src/plugins/condition/components/inspector/ConditionalStylesView.vue:209]: may wanna use `if (isEditing) {` instead to be safe, as `this.isEditing` ll get new value after `this.$nextTick(`

Comment (deeptailor) [reply src/plugins/condition/components/inspector/ConditionalStylesView.vue:209]: Should not need to, $nextTick is used when you want to wait for the dom update before doing something. In this case we are only setting the internal isEditing value.

**Comment by deeptailor** [comment]
@charlesh88 @akhenry - this pr seems like a good candidate to merge before VISTA testing cycle, what do you think?

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue (hunk 1 above)
Comment (nikhilmandlik) [reply src/plugins/condition/components/inspector/ConditionalStylesView.vue:209]: ahh, right.

**Comment by charlesh88** [comment]
> @charlesh88 @akhenry - this pr seems like a good candidate to merge before VISTA testing cycle, what do you think?

Agree. Low risk and provides useful functionality.

**Comment by akhenry** [comment]
My instinct here is no, it's very late in the game at this point. This is not a small change, it's 150+ LOC, and while useful it's not mission critical, it's a nice to have.

That said, I will compromise and agree to merge if we can testathon this. Maybe a more limited testathon, two or three people? Focused just on Conditionals.

**Comment by shefalijoshi** [comment]
> My instinct here is no, it's very late in the game at this point. This is not a small change, it's 150+ LOC, and while useful it's not mission critical, it's a nice to have.
> 
> That said, I will compromise and agree to merge if we can testathon this. Maybe a more limited testathon, two or three people? Focused just on Conditionals.

Note that this change has the diff from the preview-condition branch as well since that's not merged yet. The highlight condition in condition set view change itself is very small.

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue [hunk 2]
```diff
@@ -287,16 +304,23 @@ export default {
...
+            if (this.stopProvidingTelemetry) {
+                this.stopProvidingTelemetry();
+            }
         },
         updateDomainObjectItemStyles(newItems) {
             //check that all items that have been styles still exist. Otherwise delete those styles
             let domainObjectStyles =  (this.domainObject.configuration && this.domainObject.configuration.objectStyles) || {};
             let itemsToRemove = [];
             let keys = Object.keys(domainObjectStyles);
+            //TODO: Need an easier way to find which properties are itemIds
```
Comment (akhenry) [inline src/plugins/condition/components/inspector/ConditionalStylesView.vue:316]: I think the difficulty here is arising from `domainObject.configuration.objectStyles` being used to store more than just styles. `defaultConditionId`, `selectedConditionId`, `conditionSetIdentifier` should be stored elsewhere. Out of scope for this change, perhaps a followup?

Comment (shefalijoshi) [reply src/plugins/condition/components/inspector/ConditionalStylesView.vue:316]: I don't think they should be stored elsewhere as these properties are only relevant if objectStyles (static or conditional) have been applied to a domainObject.

**Comment by deeptailor** [comment]
@akhenry - I have approved this PR, you have a pending review, could you close you review and merge?

**Comment by charlesh88** [comment]
Testing notes:
1. Make or edit a Condition Set. 
1. Add telemetry that evaluates in an expected way (like a sine wave).
1. Add at least two conditions that will be matched in a predictable way.
1. While editing the set, verify that the winning condition is visually highlighted as expected.
1. Save the set and exit to browse mode. Again, verify that the winning condition is visually highlighted as expected.

**Comment by jvigliotta** [comment]
Verified 5/12/2020

**Comment by charlesh88** [comment]
Verified fixed testathon 05-12-20.

**Pull request merged**